resources/lib/kodiwrapper.py

```diff
@@ -377,6 +381,20 @@ def get_global_setting(setting):
         json_result = xbmc.executeJSONRPC('{"jsonrpc": "2.0", "method": "Settings.GetSettingValue", "params": {"setting": "%s"}, "id": 1}' % setting)
         return json.loads(json_result).get('result', dict()).get('value')
 
+    def notify(self, sender, message, data):
+        ''' Send a notification to Kodi using JSON RPC '''
+        import json
```

Comment:
This whole function could become:
```python
    def notify(self, sender, message, data):
        ''' Send a notification to Kodi using JSON RPC '''
        from binascii import hexlify
        import json
        result = self.jsonrpc(method='NotifyAll', params=dict(
            sender='%s.SIGNAL' % sender,
            message=message,
            data='["%s"]' % to_unicode(hexlify(json.dumps(data).encode())),
        ))
        if result.get('result') != 'OK':
            self.log_error('Failed to send notification: {error_message}', error_message=result.get('er
            return False
        self.log('Succesfully sent notification', 'Verbose')
        return True
```

Reply:
I moved the `'.SIGNAL'` and data encoding/hexlify parts back to vrtplayer because I want to be able to use `notify` with "normal" senders and json data. (for future use)
The Up Next add-on is using a specific notification format used in https://github.com/ruuk/script.module.addon.signals
I didn't want to add another dependency to the VRT NU add-on so I implemented the `notify` function without using `script.module.addon.signals`.